rep-0150.rst

```diff
@@ -0,0 +1,206 @@
+REP: 150
+Title: ROS Melodic and Newer Metapackages
+Author: Tully Foote <tfoote@openrobotics.org>, Chris Lalancette <clalancette@openrobotics.org>
+Status: Active
+Type: Informational
+Content-Type: text/x-rst
+Created: 03-Nov-2017
+Post-History:
+
+
+Abstract
+========
+
+This REP describes the metapackages for the ROS Melodic release and for releases that follow until otherwise replaced by a similar update REP.
+This REP updates REP 131 [3]_ and REP 142 [5]_ for changes in packages.
+
+
+Motivation
+==========
+
+For a discussion on the general motivation and role of variants,
+please see REP 108 [1]_.
+
+As of ROS Hydro we have completed the transition to using
+metapackages instead of variants.
+For more information on metapackages see REP 127 [2]_.
+
+This document covers the common metapackages used for recommended
+installations.
+It is expressed in the ros/metapackages repository on github.com
+[4]_.
+
+This REP proposes the same entrypoints as REP 108 and merely updates
+the variant definitions to reflect the organizational changes in ROS
+stacks.
+In particular, in ROS Melodic the `robot` metapackage is contracted to remove
+`robot_model` and instead explicitly list the necessary packages.
+
+The definitions in this REP are more focused and concise for clearer
+purpose, and do not list all transitive dependencies explicitly.
+
+
+Specification
+=============
+
+End-user entry points
+---------------------
+
+We define the same three main entry points for ROS users as REP 108
+[1]_.
+
+ * desktop_full (recommended)
+ * desktop
+ * ros_base
+
+Metapackages
+------------
+
+ROS Core
+''''''''
+
+The `ros_core` metapackage composes the core communication protocols.
+It may not contain any GUI dependencies.
+
+::
+
+ - ros_core:
+      extends: [ros, ros_comm]
```

Comment:
> does extends refers to variants/packages or repositories?

This line doesn't make any sense and needs to be removed. Probably from REP 142 too.

Reply:
So the answer is "it applies to packages/variants and not repositories" ?

What you are suggesting is to remove that line and to add ros_comm and ros to the list of packages? is that correct?

Reply:
> So the answer is "it applies to packages/variants and not repositories" ?

All other variants list variant names in the `extends`. 

> What you are suggesting is to remove that line and to add ros_comm and ros to the list of packages? is that correct?

Yes.

Reply:
So now that ros and ros_comm are listed as packages (and not repositories). Do we need to keep `std_srvs` and `rosgraph_msgs` listed given that they are dependencies of `ros_comm`?

Reply:
> Do we need to keep `std_srvs` and `rosgraph_msgs` listed given that they are dependencies of `ros_comm`?


With the same argument you could remove almost everything from the list, e.g. `catkin` since everything else depends on it. The goal is to have a complete list which answers the question what is in a variant. This needs to be obvious without starting a dependency traversal for packages / repos which are not listed here explicitly.

Reply:
Ok so the idea is that we should list all packages in a variant, and if there is a metapackage in their source repository, there is no need to list the dependencies of that metapackage, as long as they are hosted in the same repository.
Is that a good summary?

Reply:
I would say the list enumerates ROS packages / repositories.